lib/libshare/os/linux/nfs.c

```diff
@@ -23,34 +23,34 @@
  * Copyright (c) 2002, 2010, Oracle and/or its affiliates. All rights reserved.
  * Copyright (c) 2011 Gunnar Beutner
  * Copyright (c) 2012 Cyril Plisko. All rights reserved.
+ * Copyright (c) 2019, 2020 by Delphix. All rights reserved.
  */
 
+#include <dirent.h>
 #include <stdio.h>
 #include <string.h>
 #include <strings.h>
-#include <fcntl.h>
 #include <errno.h>
+#include <sys/file.h>
+#include <sys/stat.h>
+#include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
 #include <libzfs.h>
 #include <libshare.h>
 #include "libshare_impl.h"
 
-static boolean_t nfs_available(void);
+#define	ZFS_EXPORTS_DIR			"/etc/exports.d"
+#define	ZFS_EXPORTS_FILE		ZFS_EXPORTS_DIR"/zfs.exports"
+#define	ZFS_EXPORTS_FILE_TEMP	ZFS_EXPORTS_DIR"zfs.exports.XXXXXXXX"
 
 static sa_fstype_t *nfs_fstype;
 
-/*
- * nfs_exportfs_temp_fd refers to a temporary copy of the output
- * from exportfs -v.
- */
-static int nfs_exportfs_temp_fd = -1;
-
 typedef int (*nfs_shareopt_callback_t)(const char *opt, const char *value,
     void *cookie);
 
-typedef int (*nfs_host_callback_t)(const char *sharepath, const char *host,
-    const char *security, const char *access, void *cookie);
+typedef int (*nfs_host_callback_t)(const char *sharepath, const char *filename,
+    const char *host, const char *security, const char *access, void *cookie);
 
 /*
  * Invokes the specified callback function for each Solaris share option
```

Comment:
Are these really `Solaris` share options?

Reply:
Yes, the `sharenfs` property still uses the Solaris option in linux and converts them to something that can be used by exportfs. I don't think all Solaris options are supported.